app/javascript/packages/document-capture/components/form-steps.jsx

```diff
@@ -198,6 +199,7 @@ function FormSteps({
 
   return (
     <form ref={formRef} onSubmit={toNextStep}>
+      {Object.keys(values).length > 0 && <PromptOnNavigate />}
```

Comment:
Would this still trigger if a field had an empty value or a falsy value? Do we want to check inside the values?

Reply:
> Would this still trigger if a field had an empty value or a falsy value? Do we want to check inside the values?

It would be accounted for, yes. Perhaps counterintuitively, this also applies to `undefined` as well, as long as it's an explicit property of the object (`Object.keys({ foo: undefined }).length === 1`).

I guess there could be a follow-on question of: Would we want to show the prompt if the user selects a value, then chooses to unset it, reverting back to an effectively initial state. In those cases, `Object.keys(values)` would still be `> 0`, so we'd still be prompting if they tried to navigate. I think this is pretty reasonable to expect, or at least not unreasonable enough to jump through hoops to accommodate.